theia-tree-editor/src/browser/tree-editor-widget.tsx

```diff
@@ -120,7 +121,7 @@ export abstract class BaseTreeEditorWidget extends BaseWidget implements Saveabl
      *
      * @param node The tree node to delete
      */
-    protected abstract deleteNode(node: Readonly<TreeEditor.Node>): void;
+    protected abstract deleteNode(node: Readonly<TreeEditor.Node>): MaybePromise<void>;
```

Comment:
I'd just make this a normal `Promise`. It makes sense that this is handled asynchronously anyway.

Reply:
The big advantage of MaybePromise is to support legacy implementations without having to change them.
The coffee editor is a good well-known example, and I don't think it would be impacted with MaybePromise.
https://github.com/eclipse-emfcloud/coffee-editor/blob/master/web/coffee-editor-extension/src/browser/coffee-tree/coffee-tree-editor-widget.tsx
On the other hand, Promise would force it and all users who used it as a reference implementation to update their code.

It doesn't bother me to change ResourceTreeEditorWidget which provides a default implementation probably not overridden, but forcing all other implementers to change their implementation from sync to async looks like a big deal to me.
(this same conversation is valid for all similar MaybePromise => Promise suggestions)

Reply:
OK for changes.